ocaml/xapi/xapi_host.ml

```diff
@@ -1124,10 +1124,52 @@ let crl_list ~__context ~host =
   Certificates.(local_list CRL)
 
 let certificate_sync ~__context ~host =
-  Certificates.local_sync()
+  Certificates.local_sync ()
 
 let get_server_certificate ~__context ~host =
-  Certificates.get_server_certificate()
+  Certificates.get_server_certificate ()
+
+let install_server_certificate ~__context ~host ~certificate ~private_key ~certificate_chain =
+  if Db.Pool.get_ha_enabled ~__context ~self:(Helpers.get_pool ~__context) then
+    raise Api_errors.(Server_error (ha_is_enabled, []));
+
+  let current_server_certs = Db.Certificate.get_all_records ~__context
```

Comment:
what if there are no certs?

Reply:
Right now we don't care as this is not used for installing the self-signed cert from the first boot. When adding the emergency call we'll see what needs to happen here.